### PR TITLE
add handler name to request as endpoint

### PR DESCRIPTION
**Pull request by r0fls**
Fixes: #977

### CODE REVIEW

File: docs/sanic/request_data.md [hunk 1]
```diff
@@ -120,3 +120,19 @@ args.get('titles') # => 'Post 1'
 
 args.getlist('titles') # => ['Post 1', 'Post 2']
 ```
+
+## Accessing the handler name with the request.endpoint attribute
+
+The `request.endpoint` attribute holds the handler's name. For instance, the below
+route will return "hello".
```
Comment (messense) [inline docs/sanic/request_data.md]: I think in a blueprint, it should also include the blueprint's name? For example: `foo.hello`, `bar.hello`

Comment (r0fls) [reply docs/sanic/request_data.md]: I like that idea

**Comment by r0fls** [comment]
@messense if you see a better way of implementing this with the blueprint name, let me know. I'm not fond of the if statement, as it will be executed on each request.

**Comment by messense** [comment]
Can we make `request.endpoint` a lazy property on `request` class? Ideally it should only be constructed when accessing it.

File: sanic/app.py [hunk 2]
```diff
@@ -247,6 +247,12 @@ def websocket(self, uri, host=None, strict_slashes=None,
         def response(handler):
             async def websocket_handler(request, *args, **kwargs):
                 request.app = self
+                if not getattr(handler, '__blueprintname__', False):
```
Comment (profuel) [inline sanic/app.py]: will you please make this (and used in converted_response_type) portion a function.
Also I would drop 'else' statement there, so it looks like:
def _get_endpoint(handler):
      if not getattr(handler, '__blueprintname__', False):
         return handler.__name__
      return getattr(handler,
                                       '__blueprintname__',
                                      '') + handler.__name__

and used as 
request.endpoint = _get_endpoint(handler)

**Comment by ahopkins** [comment]
@r0fls Are we keeping this? Can you either update to resolve conflicts or close it?

**Comment by ahopkins** [comment]
I updated this PR to current master, made a few tweaks, added some tests, but was not allowed to push to this branch. Therefore, I am closing and will submit a new PR.

**Pull request closed without merging**